plugins/pulp_rpm/plugins/importers/yum/depsolve.py

```diff
@@ -421,8 +432,6 @@ def get_requirements(self, units):
         :rtype:     generator
         """
         for segment in paginate(units):
-            unit_ids = [unit.id for unit in segment]
-            fields = ['requires', 'id']
-            for result in models.RPM.objects.filter(id__in=unit_ids).only(*fields):
-                for require in result.requires or []:
+            for unit in segment:
+                for require in unit.requires or []:
```

Comment:
Prior to the mongoengine conversion, I suspect this code took a list of non-unit objects, and did a query against the old RPM "model" to coerce them to units in order to get at the requires metadata. Since we already have hydrated unit instances being yielded by the units generator, we don't need to go back to the database, that work is already being done. So now it just paginates the unit generator, and returns Requirements for each unit in the paginated segments.

At least that's how I think this works...I would appreciate a second opinion on this.